modules/gmc-agent/src/main/java/org/gridgain/service/tracing/RetryableSender.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2019 GridGain Systems, Inc. and Contributors.
+ *
+ * Licensed under the GridGain Community Edition License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.gridgain.com/products/software/community-edition/gridgain-community-edition-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gridgain.service.tracing;
+
+import org.apache.ignite.IgniteLogger;
+
+import java.util.concurrent.LinkedBlockingDeque;
+
+/**
+ * Retryable sender with limited queue.
+ */
+public abstract class RetryableSender<T> implements Runnable {
```

Comment:
May be instead of `abstract  class` we can implement class that accept Runnable / Closure as argument?
Usage will be like: `new RetryableSender<SpanBatch>(log, QUEUE_CAP, () -> {some_code});`

Reply:
Done.